### PR TITLE
[backend/frontend] Improve global search in files (#1483)

### PR DESCRIPTION
**Pull request by SouadHadjiat**
### Proposed changes

* Adapt layout of global search to be able to search in entities (knowledge) or files
* Display a message in files search with an action to configure file indexing if it is not running
* Display a button in knowledge search if there is less than 5 results to extend the search to indexed files
* Display "Last indexation" date in File indexing configuration page (settings => file indexing)

### Related issues

Final PR for :
* https://github.com/OpenCTI-Platform/opencti/issues/1483

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion) => will be done in this PR : https://github.com/OpenCTI-Platform/docs/pull/64/
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by labo-flg** [comment]
The top-right button that opens the advanced search panel redirects then to `http://localhost:3000/dashboard/search` 
You might either redirect to `http://localhost:3000/dashboard/search/knowledge` or event better, add a swipe rule in the router to redirect `http://localhost:3000/dashboard/search/*` to `http://localhost:3000/dashboard/search/knowledge`

File: opencti-platform/opencti-front/src/private/components/SearchRoot.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            )}
+          />
+          <BoundaryRoute
+            exact
+            path="/dashboard/search/files/:keyword"
+            render={(routeProps) => (
+              <SearchIndexedFiles {...routeProps} />
+            )}
+          />
+        </Switch>
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/SearchRoot.tsx:83]:
```suggestion
          {/* swipe rule: redirects to knowledge */}
          <Redirect to="/dashboard/search/knowledge" />
        </Switch>
```

**Comment by labo-flg** [comment]
Tested ok !

Few question/remarks : 
* file search is whole word only, do you confirm ?
* I have a lot of warnings about event emitter memory leaks in backend logs

```
(node:25540) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 101 abort listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:25540) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 101 abort listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit

```

**Comment by marieflorescontact** [comment]
> * file search is whole word only, do you confirm ?

yes

**Review by marieflorescontact — APPROVED** [review]

**Pull request merged**